src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+        ]);
+    }
+
+    public function testGetClientsUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetNonexistentClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => $clientName
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_2']);
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $updatedName = $this->createClientName();
+        $updateClientJson = ClientMother::withAllParameters(
+            $updatedName,
+            1,
+            "description updated",
+            true,
+            5,
+            [$scriptId],
+            [],
+            -1,
+            "rsync long arguments updated",
+            "rsync short arguments updated",
+            "ssh arguments updated",
+            "root@172.0.0.2"
+        );
+        $httpClient->request('PUT', $iri, [
+            'json' => $updateClientJson
+        ]);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($updateClientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testUpdateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_3']);
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'description' => 'description updated',
+                'isActive' => true,
+                'maxParallelJobs' => -1,
+                'name' => $clientName,
+                'owner' => 1,
+                'quota' => -1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
```

Comment:
Supposing you improve/change ClientMother to return not just the JSON bat a class with IRI, Context, and JSON what about something like
`
$existingClient = ClientMother::getExistingClient();
$json = $existingClient->getJson();
$jsont[ClientMother::PARAM_MAX_PARALLEL_JOBS] = -1;
$httpClient->request('PUT', $existingClient->getIri(), [ 'json' => $json]);
`